### ISSUE TITLE
Escape mysql passwords

### ISSUE THREAD

**Issue by Germanaz0** [holland-backup/holland#124]
When I've a password that ends with ; I've to escape using \; but still does not work properly.

**Comment by abg** [comment]
Can you provide some details on your environment?  Particularly providing holland --version, mysqldump --version, and the version of libmysqlclient you are using would be useful.  I assume the error you are seeing is some variant of "1045: Access denied for user ...". Any relevant logs you are able to provide would be useful.

I wasn't able to reproduce this on either the v1.0.10 tag or master branch using the following config:

```
$ cat etc/holland/backupsets/default.conf
[holland:backup]
plugin = mysqldump
auto-purge-failures = no

[mysqldump]

[mysql:client]
user = backups
password = "foo ;"
```

I also tried with the password only specified in the ~/.my.cnf rather than in the holland config:

```
$ cat ~/.my.cnf
[client]
user = backups
password = foo ;
$

$ cat etc/backupsets/default.conf
[holland:backup]
plugin = mysqldump
auto-purge-failures = no
$
```

holland generates a my.cnf under the backup directory to avoid passing the password on the commandline.  I don't see an obvious issue with how holland 1.0.10 is generating the my.cnf, but perhaps some difference in your environment is leading to a problem.

**Comment by Germanaz0** [comment]
Thanks for the reply, I'm going to send you all the informantion below:

**Holland version:** v1.0.10
**mysqldump**  Ver 10.13 Distrib 5.1.73, for unknown-linux-gnu (x86_64)
**OS**: Centos 6

If I escape the password without using \; I get the error 1045: Access denied for user ..."

But if I don't escape I get an awefull python error

```shell
Uncaught exception while running command 'bk': TypeError('not all arguments converted during string formatting',)
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/holland-1.0.10-py2.6.egg/holland/core/command/command.py", line 209, in dispatch
    return self.run(self.optparser.prog, opts, *args)
  File "/usr/lib/python2.6/site-packages/holland-1.0.10-py2.6.egg/holland/commands/backup.py", line 102, in run
    runner.backup(name, config, opts.dry_run)
  File "/usr/lib/python2.6/site-packages/holland-1.0.10-py2.6.egg/holland/core/backup/base.py", line 120, in backup
    estimated_size = self.check_available_space(plugin, spool_entry, dry_run)
  File "/usr/lib/python2.6/site-packages/holland-1.0.10-py2.6.egg/holland/core/backup/base.py", line 213, in check_available_space
    estimated_bytes_required = plugin.estimate_backup_size()
  File "/usr/lib/python2.6/site-packages/holland.backup.mysqldump-1.0.10dev-py2.6.egg/holland/backup/mysqldump/plugin.py", line 123, in estimate_backup_size
    self.schema.refresh(db_iter=db_iter, tbl_iter=tbl_iter)
  File "/usr/lib/python2.6/site-packages/holland.lib.mysql-1.0.10dev-py2.6.egg/holland/lib/mysql/schema/base.py", line 135, in refresh
    for table in tbl_iter(database.name):
  File "/usr/lib/python2.6/site-packages/holland.lib.mysql-1.0.10dev-py2.6.egg/holland/lib/mysql/schema/base.py", line 293, in __call__
    for metadata in self.client.show_table_metadata(database):
  File "/usr/lib/python2.6/site-packages/holland.lib.mysql-1.0.10dev-py2.6.egg/holland/lib/mysql/client/base.py", line 198, in show_table_metadata
    return self._show_table_metadata51(database)
  File "/usr/lib/python2.6/site-packages/holland.lib.mysql-1.0.10dev-py2.6.egg/holland/lib/mysql/client/base.py", line 181, in _show_table_metadata51
    cursor.execute(sql, (database))
  File "/usr/lib/python2.6/site-packages/MySQL_python-1.2.5-py2.6-linux-x86_64.egg/MySQLdb/cursors.py", line 187, in execute
    query = query % tuple([db.literal(item) for item in args])
TypeError: not all arguments converted during string formatting
```

**Comment by abg** [comment]
This is issue #106.  Holland 1.0.10 is not compatible with MySQL-python 1.2.5.  As a quick solution installing plugins/holland.lib.mysql from the holland master branch includes a fix for this (per issue #106).  You might use standard packages on CentOS using rpms from EPEL and use MySQL-python 1.2.3 as shipped with RHEL (and the default for most other distributions) where this is not a problem.  You would need to remove the existing MySQL-python 1.2.5, however.

The holland project needs to release a new version that includes this fix.

**Comment by m00dawg** [comment]
If a new version is going to come out, it'd be a good time to finish 
that doc refactor I was doing (but wanted some feedback on). I'd like to 
see what folks think about the rsync plugin, but that's not required to 
update the docs.

**Comment by abg** [comment]
This was fixed in v1.0.12
